test/gtest/ucp/ucp_test.cc

```diff
@@ -270,44 +270,59 @@ ucp_test::request_progress(void *req, const std::vector<entity*> &entities,
     return status;
 }
 
-ucs_status_t ucp_test::request_process(void *req, int worker_index, bool wait,
-                                       bool wakeup)
+ucs_status_t ucp_test::request_check_status(void *req, bool release)
 {
     if (req == NULL) {
         return UCS_OK;
     }
 
     if (UCS_PTR_IS_ERR(req)) {
-        ucs_error("operation returned error: %s",
-                  ucs_status_string(UCS_PTR_STATUS(req)));
         return UCS_PTR_STATUS(req);
     }
 
+    ucs_assert(UCS_PTR_IS_PTR(req));
+    ucs_status_t status = ucp_request_check_status(req);
+    if (release && (status != UCS_INPROGRESS)) {
+        ucp_request_release(req);
+    }
+
+    return status;
+}
+
+ucs_status_t ucp_test::request_process(void *req, int worker_index, bool wait,
+                                       bool wakeup, ucs_log_level_t log_level)
+{
     ucs_time_t deadline = ucs::get_deadline();
     while (wait &&
+           (UCS_PTR_IS_PTR(req)) &&
            (ucp_request_check_status(req) == UCS_INPROGRESS) &&
            (ucs_get_time() < deadline)) {
         check_events(entities(), wakeup, worker_index);
     }
 
-    ucs_status_t status = ucp_request_check_status(req);
-    if (status == UCS_INPROGRESS) {
+    ucs_status_t status = request_check_status(req, true);
+    if (UCS_STATUS_IS_ERR(status)) {
+        /* UCS errors are suppressed in case of error handling tests */
+        ucs_log(log_level, "request %p completed with error %s", req,
+                ucs_status_string(status));
+        return status;
+    } else if (status == UCS_OK) {
+        return UCS_OK;
+    } else {
```

Comment:
1. seems like duplication of the logic inside request_check_status
2. can we put this inside the loop, the replace the call to ucp_request_check_status in line 298?